server/src/frontend_api.rs

```diff
@@ -281,16 +325,98 @@ async fn post_frontend_metrics(
     Ok(HttpResponse::Accepted().finish())
 }
 
+#[utoipa::path(
+context_path = "/api",
+responses(
+(status = 202, description = "Accepted client application registration"),
+(status = 403, description = "Was not allowed to register client"),
+),
+request_body = ClientApplication,
+security(
+("Authorization" = [])
+)
+)]
+#[post("/proxy/client/register")]
+pub async fn post_proxy_register(
+    edge_token: EdgeToken,
+    connect_via: Data<ConnectVia>,
+    client_application: Json<ClientApplication>,
+    metrics_cache: Data<MetricsCache>,
+) -> EdgeResult<HttpResponse> {
+    let client_application = client_application.into_inner();
+    let updated_with_connection_info = client_application.connect_via(
+        connect_via.app_name.as_str(),
+        connect_via.instance_id.as_str(),
+    );
+    let to_write = ClientApplication {
+        environment: edge_token.environment,
+        ..updated_with_connection_info
+    };
+    metrics_cache.applications.insert(
+        ApplicationKey {
+            app_name: to_write.app_name.clone(),
+            instance_id: to_write
+                .instance_id
+                .clone()
+                .unwrap_or_else(|| ulid::Ulid::new().to_string()),
+        },
+        to_write,
+    );
+    Ok(HttpResponse::Accepted().finish())
+}
+
+#[utoipa::path(
+context_path = "/api",
+responses(
+(status = 202, description = "Accepted client application registration"),
+(status = 403, description = "Was not allowed to register client"),
+),
+request_body = ClientApplication,
+security(
+("Authorization" = [])
+)
+)]
+#[post("/frontend/client/register")]
+pub async fn post_frontend_register(
+    edge_token: EdgeToken,
+    connect_via: Data<ConnectVia>,
+    client_application: Json<ClientApplication>,
+    metrics_cache: Data<MetricsCache>,
+) -> EdgeResult<HttpResponse> {
+    let client_application = client_application.into_inner();
+    let updated_with_connection_info = client_application.connect_via(
+        connect_via.app_name.as_str(),
+        connect_via.instance_id.as_str(),
+    );
+    let to_write = ClientApplication {
+        environment: edge_token.environment,
+        ..updated_with_connection_info
+    };
+    metrics_cache.applications.insert(
+        ApplicationKey {
+            app_name: to_write.app_name.clone(),
+            instance_id: to_write
+                .instance_id
+                .clone()
+                .unwrap_or_else(|| ulid::Ulid::new().to_string()),
+        },
+        to_write,
+    );
```

Comment:
Also, same consideration as above, the code is the same as for /proxy/client/register, but perhaps it's fine